pkg/splunk/enterprise/clustermanager.go

```diff
@@ -434,3 +451,160 @@ func VerifyCMisMultisite(ctx context.Context, cr *enterpriseApi.ClusterManager,
 	}
 	return extraEnv, err
 }
+
+// upgradeScenario checks if it is suitable to update the clusterManager based on the Status of the licenseManager, returns bool, err accordingly
+func upgradeScenario(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.ClusterManager) (bool, error) {
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("upgradeScenario").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(SplunkClusterManager, cr.GetName()),
+	}
+
+	// check if the stateful set is created at this instance
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return true, nil
+	}
+
+	licenseManagerRef := cr.Spec.LicenseManagerRef
+	namespacedName = types.NamespacedName{Namespace: cr.GetNamespace(), Name: licenseManagerRef.Name}
+
+	// create new object
+	licenseManager := &enterpriseApi.LicenseManager{}
+
+	// get the license manager referred in cluster manager
+	err = c.Get(ctx, namespacedName, licenseManager)
+	if err != nil {
+		return true, nil
+	}
+
+	lmImage, err := getLicenseManagerCurrentImage(ctx, c, licenseManager)
+	if err != nil {
+		eventPublisher.Warning(ctx, "upgradeScenario", fmt.Sprintf("Could not get the License Manager Image. Reason %v", err))
+		scopedLog.Error(err, "Unable to licenseManager current image")
+		return false, err
+	}
+	cmImage, err := getClusterManagerCurrentImage(ctx, c, cr)
+	if err != nil {
+		eventPublisher.Warning(ctx, "upgradeScenario", fmt.Sprintf("Could not get the Cluster Manager Image. Reason %v", err))
+		scopedLog.Error(err, "Unable to clusterManager current image")
+		return false, err
+	}
+
+	// check conditions for upgrade
+	if cr.Spec.Image != cmImage && lmImage == cr.Spec.Image && licenseManager.Status.Phase == enterpriseApi.PhaseReady {
+		return true, nil
+	}
+
+	// Temporary workaround to keep the clusterManager method working only when the LM is ready
+	if licenseManager.Status.Phase == enterpriseApi.PhaseReady {
+		return true, nil
+	}
```

Comment:
Sorry did not understand this part, why this workaround?

Reply:
According to the initial condition, any update in the CM will only happen if there was a change in the image. To avoid this and for the situation where the LM has an image tag greater than the CM (still valid), this workaround was added.